### ISSUE TITLE
Extra Attributes get duplicated when hitting ENTER in the middle of a line

### ISSUE THREAD

**Issue by maximilianschmitt** [remirror/remirror#1652]
### Summary

When using [extra attributes](https://remirror.io/docs/concepts/extra-attributes) like in the following example, I'm expecting every paragraph to receive a unique ID:

```
  const { manager, state, getContext } = useRemirror({
    extraAttributes: [
      {
        identifiers: ["paragraph"],
        attributes: {
          id: () => _.uniqueId()
        }
      }
    ]
  });
```

However, when typing a line and inserting the cursor somewhere in the middle of that paragraph and then hitting ENTER, the following paragraph gets the same ID as the previous paragraph.

### Steps to reproduce

I've created a [minimal example on CodeSandbox](https://codesandbox.io/s/remirror-bug-duplicate-ids-7utidg?file=/src/App.js).

### Expected results

When hitting ENTER in the middle of a paragraph, the new paragraph should receive its own "Extra Attributes".

### Actual results

When hitting ENTER in the middle of a paragraph, the "Extra Attributes" of that paragraph get copied over to the new paragraph.

### Screenshot(s)

https://user-images.githubusercontent.com/3694962/169800041-f5c6725d-506e-4d33-842b-174c1af95808.mp4

**Comment by whawker** [comment]
Thanks @maximilianschmitt, good spot.

I suspect we need to add an `appendTransaction` which determines the changed nodes. If the node ID is a duplicate, and didn't exist before (by checking step mapping) we regenerate the ID.

The thing I'm unsure of, is how to determine in `appendTransaction` if the ID was generated by a unique generator, and how to use the same unique generator when there are duplicates

**Comment by maximilianschmitt** [comment]
Thanks for checking this out @whawker! I'm not very familiar with remirror/codemirror internals but as a user of the library I would have thought that these `extraAttributes` are applied to every new paragraph, even if that new paragraph was created by hitting `Enter` in the middle of a line. Or is there something special about the `id` attribute that it needs "extra treatment"? 🤔 

```
extraAttributes: [
  {
    identifiers: ["paragraph"],
    attributes: {
      id: () => _.uniqueId()
    }
  }
]
```

**Comment by whawker** [comment]
> that new paragraph was created by hitting Enter in the middle of a line

Unfortunately that's the case, when hitting Enter in the middle of a line, you don't create a new paragraph, instead you _split_ an existing paragraph into two pieces

**Comment by whawker** [comment]
Yes, this would go back to the appendTransaction method I mentioned earlier. We'd need to detect changed nodes and identify duplicate IDs. 

I'm on holiday right now, so can't look into this until next week at the earliest I'm afraid

**Comment by maximilianschmitt** [comment]
Ok great!

No problem @whawker, really appreciate this awesome project you guys are building here! Have a good holiday :)
